trinity/protocol/bcc/servers.py

```diff
@@ -0,0 +1,124 @@
+import functools
+import itertools
+from typing import (
+    cast,
+    Any,
+    Dict,
+    Iterable,
+    Set,
+    Type,
+)
+
+from cytoolz import (
+    cons,
+    sliding_window,
+    take,
+)
+
+from eth_typing import (
+    Hash32,
+)
+
+from eth_utils import (
+    to_tuple,
+)
+
+from cancel_token import CancelToken
+
+from p2p import protocol
+from p2p.peer import BasePeer
+from p2p.protocol import Command
+
+from eth.exceptions import BlockNotFound
+from eth.beacon.db.chain import BaseBeaconChainDB
+from eth.beacon.types.blocks import BaseBeaconBlock
+
+from trinity.protocol.common.servers import BaseRequestServer
+from trinity.protocol.bcc import commands
+from trinity.protocol.bcc.peer import (
+    BCCPeer,
+    BCCPeerPool,
+)
+
+
+class BCCRequestServer(BaseRequestServer):
+    subscription_msg_types: Set[Type[Command]] = {
+        commands.GetBeaconBlocks,
+    }
+
+    def __init__(self,
+                 db: BaseBeaconChainDB,
+                 peer_pool: BCCPeerPool,
+                 token: CancelToken = None) -> None:
+        super().__init__(peer_pool, token)
+        self.db = db
+
+    async def _handle_msg(self, base_peer: BasePeer, cmd: Command,
+                          msg: protocol._DecodedMsgType) -> None:
+        peer = cast(BCCPeer, base_peer)
+
+        if isinstance(cmd, commands.GetBeaconBlocks):
+            await self._handle_get_beacon_blocks(peer, cast(Dict[str, Any], msg))
+        else:
+            raise Exception("Invariant: Only subscribed to GetBeaconBlocks")
+
+    async def _handle_get_beacon_blocks(self, peer: BCCPeer, msg: Dict[str, Any]) -> None:
```

Comment:
Can you use a `mypy_extensions.TypedDict` for the `msg` to give us stronger typing and remove the need for the `cast` call.

Reply:
Nice, didn't know about `TypeDict`, seems pretty useful! I added it, but here it didn't help a lot unfortunately: It only got rid of the cast for `max_blocks`, we still need to first cast the message to the `TypeDict`.